modules/jdbc/src/main/java/org/testcontainers/jdbc/ConnectionUrl.java

```diff
@@ -0,0 +1,209 @@
+        this.url = Objects.requireNonNull(url, "Connection URL cannot be null");
+    }
+
+    public String getDatabaseType() {
+        return Objects.requireNonNull(this.databaseType, "Database Type cannot be null. Have you called parseUrl() method?");
+    }
+
+
+    /**
+     * This is a part of the connection string that may specify host:port/databasename.
+     * It may vary for different clients and so clients can parse it as needed.
+     *
+     * @return
+     */
+    public String getDbHostString() {
+        return Objects.requireNonNull(this.dbHostString, "Database Host String cannot be null. Have you called parseUrl() method?");
+    }
+
+    public static boolean accepts(final String url) {
+        return url.startsWith("jdbc:tc:");
+    }
+
+    public void parseUrl() {
+        /*
+        Extract from the JDBC connection URL:
+         * The database type (e.g. mysql, postgresql, ...)
+         * The docker tag, if provided.
+         * The URL query string, if provided
+       */
+        Matcher urlMatcher = Patterns.URL_MATCHING_PATTERN.matcher(this.getUrl());
+        if (!urlMatcher.matches()) {
+            //Try for Oracle pattern
+            urlMatcher = Patterns.ORACLE_URL_MATCHING_PATTERN.matcher(this.getUrl());
+            if (!urlMatcher.matches()) {
+                throw new IllegalArgumentException("JDBC URL matches jdbc:tc: prefix but the database or tag name could not be identified");
+            }
+        }
+        databaseType = urlMatcher.group(1);
+
+        imageTag = Optional.ofNullable(urlMatcher.group(3)).orElse("latest");
+
+        //String like hostname:port/database name, which may vary based on target database.
+        //Clients can further parse it as needed.
+        dbHostString = urlMatcher.group(4);
+
+        //In case it matches to the default pattern
+        Matcher dbInstanceMatcher = Patterns.DB_INSTANCE_MATCHING_PATTERN.matcher(dbHostString);
+        if (dbInstanceMatcher.matches()) {
+            databaseHost = Optional.of(dbInstanceMatcher.group(1));
+            databasePort = Optional.ofNullable(dbInstanceMatcher.group(3)).map(value -> Integer.valueOf(value));
+            databaseName = Optional.of(dbInstanceMatcher.group(4));
+        }
+
+        queryString = Optional.ofNullable(urlMatcher.group(5));
+        getQueryParameters();
+
+        Matcher matcher = Patterns.INITSCRIPT_MATCHING_PATTERN.matcher(this.getUrl());
+        if (matcher.matches()) {
+            initScriptPath = Optional.ofNullable(matcher.group(2));
+        }
+
+        Matcher funcMatcher = Patterns.INITFUNCTION_MATCHING_PATTERN.matcher(this.getUrl());
+        if (funcMatcher.matches()) {
+            initFunction = Optional.of(new InitFunctionDef(funcMatcher.group(2), funcMatcher.group(4)));
+        }
+
+        Matcher daemonMatcher = Patterns.DAEMON_MATCHING_PATTERN.matcher(this.getUrl());
+        inDaemonMode = daemonMatcher.matches() ? Boolean.parseBoolean(daemonMatcher.group(2)) : false;
```

Comment:
IntelliJ is telling me that this can be simplified to:
```
inDaemonMode = daemonMatcher.matches() && Boolean.parseBoolean(daemonMatcher.group(2));
```

Reply:
This has been refactored now.